app/HutchManager/Data/Entities/DataSource.cs

```diff
@@ -4,4 +4,6 @@ public class DataSource
 {
   public string Id { get; set; } = string.Empty;
   public DateTimeOffset LastCheckin { get; set; }
+
+  public List<Agent>? Agents { get; set; } = new();
```

Comment:
```suggestion
  public List<Agent> Agents { get; set; } = new();
```

Lists on entities (or in 95% of classes tbh) never need to be nullable. Because an empty list (not `null`) is how you represent no items.